src/runtime_src/core/pcie/emulation/cpu_em/generic_pcie_hal2/swscheduler.h

```diff
@@ -0,0 +1,206 @@
+#ifndef _SW_SCHEDULER_H_
+#define _SW_SCHEDULER_H_
+
+#include <list>
+#include <mutex>
+#include <cmath>
+#include <cstdint>
+#include <queue>
+#include "ert.h"
+
+#define XOCL_U32_MASK 0xFFFFFFFF
+
+#define	MAX_SLOTS	128
+#define MAX_CUS		128
+#define MAX_U32_SLOT_MASKS (((MAX_SLOTS-1)>>5) + 1)
+#define MAX_U32_CU_MASKS (((MAX_CUS-1)>>5) + 1)
+
+namespace xclcpuemhal2 {
+  class CpuemShim;
+  class xocl_cmd;
+  class SWScheduler;
+  class exec_core;
+
+  struct client_ctx 
+  {
+    int		trigger;
+    std::mutex mLock;
+  };
+
+  class xocl_sched
+  {
+    public:
+      pthread_t                   scheduler_thread;
+      pthread_mutex_t             state_lock;
+      pthread_cond_t              state_cond;

```

Comment:
No, please use std::thread library.  Do *not* use pthread APIs directly.

Reply:
changed to std::thread library